che-theia-factory/src/browser/factory-theia-client.ts

```diff
@@ -85,22 +137,39 @@ export class FactoryTheiaClient implements FrontendApplicationContribution {
                 (repo: Repository) => {
                     this.messageService.info(`Project ${projectPath} successfully cloned.`);
                     if (source.parameters['branch']) {
-                        const options: Git.Options.Checkout.CheckoutBranch = { branch: source.parameters['branch'] };
+                        const options: Git.Options.Checkout.CheckoutBranch = {branch: source.parameters['branch']};
                         this.git.checkout(repo, options);
                     }
+                    return Promise.resolve();
                 }
             ).catch((error) => {
                 console.error(`Couldn't clone ${source.location} to ${projectPath}... ${error}`);
```

Comment:
Minor, but such case: clone could be done successfully, but than error can be thrown on checkout operation. Message service always displays  error message "Couldn't clone...", but in this case it should only notify about checkout fail.

Reply:
Good catch. Done